src/main/java/net/javadiscord/javabot/systems/help/HelpManager.java

```diff
@@ -167,12 +163,35 @@ private boolean hasMemberHelperRole(@NotNull Guild guild, @Nullable Member membe
 	}
 
 	private @NotNull List<Member> getPostHelpers() {
-		List<Message> messages = ForumHelpListener.HELP_POST_MESSAGES.get(postThread.getIdLong());
+		List<Message> messages = HelpListener.HELP_POST_MESSAGES.get(postThread.getIdLong());
 		if (messages == null) return List.of();
 		return messages.stream()
 				.filter(m -> m.getMember() != null && m.getAuthor().getIdLong() != postThread.getOwnerIdLong())
 				.map(Message::getMember)
 				.distinct()
 				.toList();
 	}
+
+	/**
+	 * Calculates the experience for each user, based on the messages they sent.
+	 *
+	 * @param messages The list of {@link Message}s.
+	 * @param ownerId The owner id.
+	 * @param config The {@link HelpConfig}, containing some static info for the calculation.
+	 * @return A {@link Map}, containing the users' id as the key, and the amount of xp as the value.
+	 */
+	public static Map<Long, Double> calculateExperience(List<Message> messages, long ownerId, HelpConfig config) {
+		Map<Long, Double> experience = new HashMap<>();
+		if (messages == null || messages.isEmpty()) return Map.of();
+		for (User user : messages.stream().map(Message::getAuthor).collect(Collectors.toSet())) {
```

Comment:
Why not just use `Collectors.groupingby()` and getting the count instead of getting all messages?